src/t8_geometry/t8_geometry_implementations/t8_geometry_linear.cxx

```diff
@@ -54,6 +56,356 @@ t8_geometry_linear::t8_geom_evaluate_jacobian (t8_cmesh_t cmesh, t8_gloidx_t gtr
+{
+  /* Let p0, p1, p2, p3 be the four points.
+   * The four points are coplanar if the normal vectors to the triangles
+   * p0, p1, p2 and p0, p2, p3 are pointing in the same direction.
+   *
+   * We build the vectors A = p1 - p0, B = p2 - p0 and C = p3 - p0.
+   * The normal vectors to the triangles are n1 = A x B and n2 = A x C.
+   * These are pointing in the same direction if their cross product is 0.
+   * Hence we check if || n1 x n2 || < tolerance. */
+
+  /* A = p1 - p0 */
+  double A[3];
+  t8_vec_axpyz (p_0, p_1, A, -1);
+
+  /* B = p2 - p0 */
+  double B[3];
+  t8_vec_axpyz (p_0, p_2, B, -1);
+
+  /* C = p3 - p0 */
+  double C[3];
+  t8_vec_axpyz (p_0, p_3, C, -1);
+
+  /* n1 = A x B */
+  double A_cross_B[3];
+  t8_vec_cross (A, B, A_cross_B);
+
+  /* n2 = A x C */
+  double A_cross_C[3];
+  t8_vec_cross (A, C, A_cross_C);
+
+  /* n1 x n2 */
+  double n1_cross_n2[3];
+  t8_vec_cross (A_cross_B, A_cross_C, n1_cross_n2);
+
+  /* || n1 x n2 || */
+  const double norm = t8_vec_norm (n1_cross_n2);
+  return norm < tolerance;
+}
+#endif
+
+void
+t8_geometry_linear::t8_geom_point_batch_inside_element (t8_forest_t forest, t8_locidx_t ltreeid,
+                                                        const t8_element_t *element, const double *points,
+                                                        const int num_points, int *is_inside,
+                                                        const double tolerance) const
+{
+  const t8_eclass_t tree_class = t8_forest_get_tree_class (forest, ltreeid);
+  t8_eclass_scheme_c *ts = t8_forest_get_eclass_scheme (forest, tree_class);
+  const t8_element_shape_t element_shape = ts->t8_element_shape (element);
+  switch (element_shape) {
+  case T8_ECLASS_VERTEX: {
+    /* A point is 'inside' a vertex if they have the same coordinates */
+    double vertex_coords[3];
+    /* Get the vertex coordinates */
+    t8_forest_element_coordinate (forest, ltreeid, element, 0, vertex_coords);
+    /* Check whether the point and the vertex are within tolerance distance
+       * to each other */
+    for (int ipoint = 0; ipoint < num_points; ipoint++) {
+      is_inside[ipoint] = t8_vertex_point_inside (vertex_coords, &points[ipoint * 3], tolerance);
+    }
+    return;
+  }
+  case T8_ECLASS_LINE: {
+    /* A point p is inside a line that is defined by the edge nodes
+     * p_0 and p_1
+     * if and only if the linear system
+     * (p_1 - p_0)x = p - p_0
+     * has a solution x with 0 <= x <= 1
+     */
+    double p_0[3], v[3];
+
+    /* Compute the vertex coordinates of the line */
+    t8_forest_element_coordinate (forest, ltreeid, element, 0, p_0);
+    /* v = p_1 */
+    t8_forest_element_coordinate (forest, ltreeid, element, 1, v);
+    /* v = p_1 - p_0 */
+    t8_vec_axpy (p_0, v, -1);
+    for (int ipoint = 0; ipoint < num_points; ipoint++) {
+      is_inside[ipoint] = t8_line_point_inside (p_0, v, &points[ipoint * 3], tolerance);
+    }
+    return;
```

Comment:
But wouldn't this also return true if the point is on the line, but not between both vertices?

Reply:
No, because the line check tests if the point can be defined by `p_0 + x * v` with `-eps < x < 1 + eps`